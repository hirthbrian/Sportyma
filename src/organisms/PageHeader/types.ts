export interface Props {
  title: string;
  date: Date;
}
