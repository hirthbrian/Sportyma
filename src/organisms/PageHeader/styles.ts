import styled from "styled-components";

export const Container = styled.View`
  align-items: center;
  padding-top: 15px;
`;
