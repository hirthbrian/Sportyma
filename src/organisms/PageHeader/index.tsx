import { format } from "date-fns";

import KanitText from "../../atoms/KanitText";
import DayPill from "../../molecules/DayPill";

import { Container } from "./styles";
import { Props } from "./types";

const PageHeader = ({ date }: Props) => {
  return (
    <Container>
      <DayPill date={date} />
      <KanitText size={20}>{format(new Date(date), "HH:mm")}</KanitText>
    </Container>
  );
};

export default PageHeader;
