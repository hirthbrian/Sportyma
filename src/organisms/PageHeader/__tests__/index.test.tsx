import React from "react";
import renderer from "react-test-renderer";

import PageHeader from "..";

const date = "2023-02-16T20:35:00+00:00" as unknown as Date;

describe("<PageHeader />", () => {
  let props;

  beforeEach(() => {
    props = {
      date,
    };
  });

  it("renders correcly", () => {
    expect(renderer.create(<PageHeader {...props} />).toJSON()).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer.create(<PageHeader {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
