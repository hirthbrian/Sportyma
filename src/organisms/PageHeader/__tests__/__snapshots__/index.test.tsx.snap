// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "paddingTop": 15,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#2559f2",
          "borderRadius": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Kanit-Regular",
            "fontSize": 16,
            "letterSpacing": 1,
            "paddingBottom": 4,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 4,
          },
          Object {
            "fontSize": 12,
          },
        ]
      }
    >
      16 FEBRUARY
    </Text>
  </View>
  <Text
    size={20}
    style={
      Array [
        Object {
          "color": "#152D4B",
          "fontFamily": "Kanit-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    21:35
  </Text>
</View>
`;
